proposals/partial-events-and-constructors.md

```diff
@@ -0,0 +1,262 @@
+     ;
+```
+
+Instance constructor declaration syntax ([§15.11.1][ctor-syntax]) is extended to allow the `partial` modifier:
+
+```diff
+ constructor_declaration
+-    : attributes? constructor_modifier* constructor_declarator constructor_body
++    : attributes? constructor_modifier* 'partial'? constructor_declarator constructor_body
+     ;
+```
+
+Note that there is [a proposal][partial-ordering] to allow the `partial` modifier anywhere among modifiers,
+rather than only as the last one (also for method, property, and type declarations).
+
+An event declaration with the `partial` modifier is said to be a *partial event declaration*
+and it is associated with one or more *partial events* with the specified names
+(note that one event declaration without accessors can define multiple events).
+
+A constructor declaration with the `partial` modifier is said to be a *partial constructor declaration*
+and it is associated with a *partial constructor* with the specified signature.
+
+A partial event declaration is said to be an *implementing declaration*
+when it specifies the `event_accessor_declarations` or it has the `extern` modifier.
+Otherwise, it is a *defining declaration*.
+
+A partial constructor declaration is said to be a *defining declaration*
+when it has the semicolon body and it lacks the `extern` modifier.
+Otherwise, it is an *implementing declaration*.
+
+```cs
+partial class C
+{
+    // defining declarations
+    partial C();
+    partial C(int x);
+    partial event Action E, F;
+
+    // implementing declarations
+    partial C() { }
+    partial C(int x) { }
+    partial event Action E { add { } remove { } }
+    partial event Action F { add { } remove { } }
+}
+```
+
+Only the defining declaration of a partial member participates in lookup
+and is considered at use sites and for emitting the metadata.
+(Except for documentation comments as detailed [below](#documentation-comments).)
+The implementing declaration signature is used when nullable-analyzing the associated bodies.
+
+A partial event or constructor:
+- May only be declared as a member of a partial type.
+- Must have one defining and one implementing declaration.
+- Is not permitted to have the `abstract` modifier.
+- Cannot explicitly implement an interface member.
+
+A partial event is not field-like ([§15.8.2][event-field-like]), i.e.:
+- It does not have any backing storage or accessors generated by the compiler.
+- It can only be used in `+=` and `-=` operations, not as a value.
+
+A defining partial constructor declaration cannot have a constructor initializer
+(`: this()` or `: base()`; [§15.11.2][ctor-init]).
+
+### Attributes
+
+The attributes of the resulting event or constructor are the combined attributes of the partial declarations.
+The attributes of each of the resulting event's accessor
+are the combined attributes of the corresponding partial declaration accessors.
+The attributes of each of the resulting constructor's parameter
```

Comment:
Perhaps "The attributes of each resulting constructor parameter ..." or "The attributes of each of the resulting constructor's parameters ..." (plural).